cfme/tests/cloud/test_cloud_init_provisioning.py

```diff
@@ -55,20 +61,50 @@ def test_provision_cloud_init(request, setup_provider, provider, provisioning,
     instance = Instance.factory(vm_name, provider, image)
 
     request.addfinalizer(instance.cleanup_on_provider)
-    # TODO: extend inst_args for other providers except EC2 if needed
     inst_args = {
         'request': {'email': 'image_provisioner@example.com',
                     'first_name': 'Image',
                     'last_name': 'Provisioner',
                     'notes': note},
-        'catalog': {'vm_name': vm_name},
-        'properties': {'instance_type': provisioning['instance_type'],
-                       'guest_keypair': provisioning['guest_keypair']},
-        'environment': {'availability_zone': provisioning['availability_zone'],
-                        'cloud_network': provisioning['cloud_network'],
-                        'security_groups': [provisioning['security_group']]},
-        'customize': {'custom_template': {'name': provisioning['ci-template']}}
+        'catalog': {
+            'vm_name': vm_name},
+        'customize': {
+            'custom_template': {'name': provisioning['ci-template']}},
+        'environment': {
+            'availability_zone': provisioning.get('availability_zone', None),
+            'security_groups': [provisioning.get('security_group', None)],
+            'cloud_network': provisioning.get('cloud_network', None),
+            'cloud_subnet': provisioning.get('cloud_subnet', None),
+            'resource_groups': provisioning.get('resource_group', None)
+        },
+        'properties': {
+            'instance_type': partial_match(provisioning.get('instance_type', None)),
+            'guest_keypair': provisioning.get('guest_keypair', None)}
     }
+    # GCE specific
+    if provider.one_of(GCEProvider):
+        recursive_update(inst_args, {
```

Comment:
you changes look sophisticated. if you need to add 2-3 items to dict, just add them instead of using complex constructions.